hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/TransferPrecompile.java

```diff
@@ -248,13 +250,19 @@ public void run(final MessageFrame frame) {
         final Map<ByteString, EntityNum> completedLazyCreates = new HashMap<>();
         for (int i = 0, n = changes.size(); i < n; i++) {
             final var change = changes.get(i);
+
             final var units = change.getAggregatedUnits();
+            final var isDebit = units < 0;
+            final var isCredit = units > 0;
+
             if (change.hasAlias()) {
                 replaceAliasWithId(change, changes, completedLazyCreates);
             }
 
-            final var isDebit = units < 0;
-            final var isCredit = units > 0;
+            // check whether the balance change is negative e.g. for "from" field, if not validate the "to" field.
```

Comment:
suggest rewriting this comment in the language of "credit"/"debit" rather than "negative"/"positive" - that's why we invented `isDebit`/`isCredit` in the first place: to move to domain concepts rather than implementation details.